integration/run.py

```diff
@@ -0,0 +1,190 @@
+        "--purge",
+        f"--sharebase={directory}-shared",
+        f"--branch={branch}",
+        url,
+        directory,
+    ]
+    subprocess.check_output(cmd)
+
+
+def tip(repo_dir):
+    """
+    Get the tip of the repo
+    """
+    cmd = ["hg", "tip", "--template={rev}"]
+    rev = subprocess.check_output(cmd, cwd=repo_dir)
+    return int(rev)
+
+
+def patch(filename, repo_dir, message):
+    """
+    Apply a locally stored patch on the repository
+    and commit the difference
+    """
+    assert os.path.isdir(repo_dir), f"Not a directory {repo_dir}"
+    path = os.path.join(ROOT, "patches", filename)
+    assert os.path.exists(path), f"Missing patch {path}"
+
+    logger.info("Applying patch", name=filename, dir=repo_dir)
+    cmd = [
+        "hg",
+        "import",
+        "--user=code-review-integration",
+        f"--message={message}",
+        path,
+    ]
+    subprocess.check_output(cmd, cwd=repo_dir)
+
+    # Load revision created
+    rev = tip(repo_dir)
+    logger.info("Committed a new revision", id=rev)
+    return rev
+
+
+def publish(repo_dir, base):
+    """
+    Publish diff on Phabricator
+    from the base of the repository
+    """
+    phab_url = taskcluster.secrets["phabricator"]["url"]
+    phab_token = taskcluster.secrets["phabricator"]["token"]
+    logger.info(
+        "Publishing a revision on phabricator", url=phab_url, local_revision=revision
+    )
+    cmd = [
+        "arc",
+        f"--conduit-uri={phab_url}",
+        f"--conduit-token={phab_token}",
+        "--no-ansi",
+        "diff",
+        "--nolint",
+        "--nounit",
+        "--excuse=integration-test",
+        "--verbatim",
+        "--create",
+        f"{base}",
+    ]
+    output = subprocess.check_output(cmd, cwd=repo_dir)
+
+    # Parse output to get the revision url
+    match = re.search(r"Revision URI: (.*)", output.decode("utf-8"), re.MULTILINE)
+    assert match is not None, "No revision found in arc output"
+
+    return match.group(1)
+
+
+def notify(message):
+    """
+    Notify admins through email
+    """
+    notify = taskcluster.get_service("notify")
+    for email in taskcluster.secrets["admins"]:
+        logger.info("Sending email", to=email)
+        notify.email(
+            {
+                "address": email,
+                "subject": "Code review integration test",
+                "content": message,
+            }
+        )
+
+
+if __name__ == "__main__":
+    logger.info("Running integration test")
+    args = configure()
+
+    # Clone NSS for a shorter time than MC
+    nss = os.path.join(args.clone_dir, "nss")
+    clone("https://hg.mozilla.org/projects/nss", nss)
+    base = tip(nss)
+
+    # Apply a specific patch on the NSS clone
+    revision = patch("nss.diff", nss, "Bug XXYYZZ - Code review integration test")
+
+    # Submit commit on Phabricator instance
+    url = publish(nss, base)
+
+    # Send notification to admins
+    notify(f"New code-review integration test: {url}")
```

Comment:
In the future, when we have the check on the publication, maybe we should only notify on failure.